test/reflect.js

```diff
@@ -0,0 +1,483 @@
+/*global describe, it, expect, require, Reflect */
+
+var exported = require('../');
+
+describe('Reflect', function () {
+  var object = {
+    something: 1,
+    _value: 0
+  };
+
+  Object.defineProperties(object, {
+    value: {
+      get: function () {
+        return this._value;
+      }
+    },
+
+    setter: {
+      set: function (val) {
+        this._value = val;
+      }
+    },
+
+    bool: {
+      value: true
+    }
+  });
+
+  var testCallableThrow = function (func) {
+    [null, undefined, 1, 'string', true, [], {}].forEach(function (item) {
+      expect(func.bind(null, item)).to.throw(TypeError);
+    });
+  };
+
+  var testPrimitiveThrow = function (func) {
+    [null, undefined, 1, 'string', true].forEach(function (item) {
+      expect(func.bind(null, item)).to.throw(TypeError);
+    });
+  };
+
+  it('is on the exported object', function () {
+    expect(exported.Reflect).to.equal(Reflect);
+  });
+
+  describe('Reflect.apply()', function () {
+    it('is a function', function () {
+      expect(typeof Reflect.apply).to.equal('function');
+    });
+
+    it('throws if target isn\'t callable', function () {
+      testCallableThrow(function (item) {
+        return Reflect.apply(item, null, []);
+      });
+    });
+
+    it('works also with redefined apply', function () {
+      expect(Reflect.apply(Array.prototype.push, [1, 2], [3, 4, 5])).to.equal(5);
+
+      function F(a, b, c) {
+        return a + b + c;
+      }
+
+      F.apply = false;
```

Comment:
The point is to have a function with a non-function apply property. If we don't guard against it, it would be possible to prevent Reflect.apply from functioning if it is implemented simply as `func.apply(context, args)`. Right?